crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2789,10 +2789,35 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = attribute;
 
         let value_ty = self.infer_expression(value);
-        value_ty
-            .member(self.db, &Name::new(&attr.id))
-            .ignore_possibly_unbound()
-            .unwrap_or(Type::Unknown)
+        match value_ty.member(self.db, &Name::new(&attr.id)) {
```

Comment:
while we're here, we can also clean this up (the method just takes `&str` these days):

```suggestion
        match value_ty.member(self.db, &attr.id) {
```